components/comment.js

```diff
@@ -1,82 +1,75 @@
-  constructor(props) {
-    super(props)
-    this.state = { toggled: false }
-    this.toggle = this.toggle.bind(this)
-  }
+import timeAgo from "../lib/time-ago";
 
-  render() {
-    const { user, text, date, comments } = this.props
-    return (
-      <div className="comment">
-        <div className="meta">
-          {/* TODO: time hydration mismatch */}
-          {user} {timeAgo(new Date(date))} ago{' '}
-          <span onClick={this.toggle} className="toggle">
-            {this.state.toggled
-              ? `[+${(this.props.commentsCount || 0) + 1}]`
-              : '[-]'}
-          </span>
-        </div>
+export default function Comment({ user, text, date, comments, commentsCount }) {
+  const [toggled, setToggled] = useState(false)
 
-        {this.state.toggled
-          ? null
-          : [
-              <div
-                key="text"
-                className="text"
-                dangerouslySetInnerHTML={{ __html: text }}
-              />,
-              <div key="children" className="children">
-                {comments.map((comment) => (
-                  <Comment key={comment.id} {...comment} />
-                ))}
-              </div>,
-            ]}
+  const toggle = () => setToggled(!toggled)
 
-        <style jsx>{`
-          .comment {
-            padding-top: 15px;
-          }
-          .children {
-            padding-left: 20px;
-          }
-          .meta {
-            font-size: 12px;
-            margin-bottom: 5px;
-          }
-          .toggle {
-            cursor: pointer;
-          }
-          .text {
-            color: #000;
-            font-size: 13px;
-            line-height: 18px;
-          }
-          /* hn styles */
-          .text :global(p) {
-            margin-top: 10px;
-          }
-          .text :global(pre) {
-            margin-bottom: 10px;
-            max-width: 900px;
-            overflow: auto;
-            padding: 2px;
-            white-space: pre-wrap;
-          }
-          .text :global(a) {
-            color: #000;
-          }
-        `}</style>
+  return (
+    <div className="comment">
+      <div className="meta">
+        {/* TODO: time hydration mismatch */}
+        {user} {timeAgo(new Date(date))} ago{' '}
+        <span onClick={toggle} className="toggle">
+          {toggled
+            ? `[+${(commentsCount || 0) + 1}]`
+            : "[-]"}
+        </span>
       </div>
-    )
-  }
 
-  toggle() {
-    this.setState({ toggled: !this.state.toggled })
-  }
+      {toggled
+        ? null
+        : [
+            <div
+              key="text"
+              className="text"
+              dangerouslySetInnerHTML={{ __html: text }}
+            />,
+            <div key="children" className="children">
+              {comments.map((comment) => (
+                <Comment key={comment.id} {...comment} />
+              ))}
+            </div>,
+          ]}
+
+      <style jsx>{`
+        .comment {
```

Comment:
Let's change these to CSS Modules?

Reply:
Unfortunately, it looks like that CSS Modules don't work with Client Components. I am preparing a PR with some failing tests that should have passed.

Reply:
https://github.com/vercel/next.js/issues/42072